jetty-server/src/main/java/org/eclipse/jetty/server/CachedContentFactory.java

```diff
@@ -234,18 +233,26 @@ private HttpContent load(String pathInContext, Resource resource, int maxBufferS
                     if (compressedContent == null || compressedContent.isValid())
                     {
                         compressedContent = null;
-                        Resource compressedResource = _factory.getResource(compressedPathInContext);
-                        if (compressedResource.exists() && compressedResource.lastModified() >= resource.lastModified() &&
-                            compressedResource.length() < resource.length())
+                        try
                         {
-                            compressedContent = new CachedHttpContent(compressedPathInContext, compressedResource, null);
-                            CachedHttpContent added = _cache.putIfAbsent(compressedPathInContext, compressedContent);
-                            if (added != null)
+                            Resource compressedResource = _factory.getResource(compressedPathInContext);
+                            if (compressedResource.exists() && compressedResource.lastModified() >= resource.lastModified() &&
+                                compressedResource.length() < resource.length())
                             {
-                                compressedContent.invalidate();
-                                compressedContent = added;
+                                compressedContent = new CachedHttpContent(compressedPathInContext, compressedResource, null);
+                                CachedHttpContent added = _cache.putIfAbsent(compressedPathInContext, compressedContent);
+                                if (added != null)
+                                {
+                                    compressedContent.invalidate();
+                                    compressedContent = added;
+                                }
                             }
                         }
+                        catch (IOException e)
```

Comment:
What would happen if we didn't catch here?    A 500 response probably, but that might be more appropriate that a null response?

Reply:
The catch exists for the speculative `_factory.getResource(compressedPathInContext)` and allows the next attempt to load below if not valid.

The old `_factory.getResource()` relied on null returning.

Let me see if I can find a good testcase for this.

Reply:
But the compressedPathInContext should not throw unless something goes horribly wrong.  It should return a resource that does not exist and we should fall through because of an exists test not because of an exception.  We REALLY don't want exceptions to be in our normal execution path!!!!